src/Controls/src/Core/Compatibility/iOS/Extensions/ToolbarItemExtensions.cs

```diff
@@ -108,6 +109,10 @@ void UpdateIconAndStyle(ToolbarItem item)
 					{
 						Image = result?.Value;
 						Style = UIBarButtonItemStyle.Plain;
+						if (item.IconImageSource is FontImageSource fontImageSource && fontImageSource.Color is not null)
```

Comment:
Will require to apply changes to fix also the secondary toolbar items. https://github.com/dotnet/maui/blob/9416968bff8156356dd98f8684e1d8f6f4965998/src/Controls/src/Core/Compatibility/iOS/Extensions/ToolbarItemExtensions.cs#L189

Reply:
The color is applied correctly for secondary toolbar items. This might be because the custom SecondaryToolbarItemContent control internally manages the icon color.